### PR TITLE
Add a dependency on symfony/asset

### PR DESCRIPTION
**Pull request by weaverryan**
Hi guys!

I originally avoided a hard dependency on `symfony/asset`, but this adds it? Why:

1) It would allow us to abandon `symfony/webpack-encore-pack`... as we can now attach the recipe to this bundle and it will bring in the needed symfony/asset package

2) Needing to use Encore inside Symfony... but NOT wanting the `symfony/asset` component is a huge edge case. And if you're in that situation, it's trivial to implement your own system for loading `entrypoints.json`.

In short, this simplifies things!

### CODE REVIEW

**Review by codayblue — APPROVED** [review]
We should merge this before #6 so that way I can resolve the conflicts in a couple places.
